### PR TITLE
User guide - Update sphinx_rtd_theme to version 0.2.4

### LINKED ISSUE #730: User guide for ci4 is not responsive on mobile.(screen width  less then 769px)
ci4-smallscreen

### PR DESCRIPTION
**Pull request by natanfelles**
Fix #730.

sphinx_rtd_theme_old can be deleted.

Please, download https://github.com/natanfelles/CodeIgniter4/tree/user_guide build and compare.

Do you know a solution for errors like this on build?

```
CodeIgniter4/user_guide_src/source/concepts/structure.rst:20: WARNING: Could not lex literal_block as "ci". Highlighting skipped.
```

### CODE REVIEW

**Comment by jim-parry** [comment]
There are a number of sphinx errors in the user guide source - not sure if this is related.
Will check out the PR.

File: user_guide_src/source/conf.py [hunk 1]
```diff
@@ -101,7 +101,7 @@
 #
 # Specifying a few options; just a starting point & we can play with it.
 html_theme_options = {
-
+    'display_version': False
```
Comment (natanfelles) [inline user_guide_src/source/conf.py]: @jim-parry Perhaps this can be `True`? https://forum.codeigniter.com/thread-69043.html

**Comment by jim-parry** [comment]
The sphinx error you encountered is present in the main branch of the project too - not related to anything you did. I need to make a sweep through and clean things up a bit, and then figure out how to integrate a sphinx build into travis-ci to prevent doc errors from being introduced :-/

File: user_guide_src/source/conf.py (hunk 1 above)
Comment (jim-parry) [reply user_guide_src/source/conf.py]: We are planning to visit user guide availability, across versions, before the 4.0 release.
That should be handled then, but not easily done using gh-pages.

**Comment by jim-parry** [comment]
Are you planning any more tweaks, or is this good to go?

**Comment by natanfelles** [comment]
As I see, the theme styles are OK. Colors according with CI and now with some extras of the new sphinx_rtd_theme.

I thought research about the lexer and study. But I do not know how it works. So for now, no further adjustments.

File: user_guide_src/source/conf.py (hunk 1 above)
Comment (natanfelles) [reply user_guide_src/source/conf.py]: Most easy method that I think is creating folders with version numbers and uploading the file already builded. 

```
/user_guide/  # Current version
/user_guide/archive/ # List of versions
/user_guide/archive/4.0.2 # Version
```

But if I'm not mistaken, I've read an article on how to build gh-pages into sub-folders... which would automate the process.

Comment (jim-parry) [reply user_guide_src/source/conf.py]: the problem isn't just gh-pages, but multiple versions in multiple repos and available in multiple forms (html, download, epub, pdf) ... leaving that for another day :)

**Pull request merged**